src/app/cursos/curso/[id]/curso-form.tsx

```diff
@@ -49,6 +51,49 @@ export const CursoForm = ({ id, onSubmit, onCancel }: Props) => {
 
   const { handleSubmit, control } = formHook;
 
+  const materias =
+    api.materia.getAll.useQuery().data?.map((item) => ({ id: item.id, label: item.nombre, data: item.anio })) ?? [];
```

Comment:
El map es una operación barata, pero si vinieran muchos resultados lo va a ejecutar todo el tiempo con los mismos datos. 

```suggestion
  const { data } = api.materia.getAll.useQuery();

  const materias = useMemo(() => {
    return data?.map((item) => ({ id: item.id, label: item.nombre, data: item.anio })) ?? [];
  }, [data]);
```

Básicamente React valida todo el componente cada vez que ocurre un "tick" eso es cada interacción con el mismo directa indirectamente. Lo que hace que se ejecute el .map todo el tiempo. Con este código, (importando el useMemo), solo se va a ejecutar cuando cambie el data.